src/Servers/Kestrel/Core/src/HttpsConnectionAdapterOptions.cs

```diff
@@ -58,7 +59,14 @@ public HttpsConnectionAdapterOptions()
         /// <summary>
         /// Specifies a callback for additional client certificate validation that will be invoked during authentication.
         /// </summary>
-        public Func<X509Certificate2, X509Chain, SslPolicyErrors, bool> ClientCertificateValidation { get; set; }
+        public Func<X509Certificate2, X509Chain, SslPolicyErrors, bool> ClientCertificateValidation {
+            get
+            {
+                return AllowAnyClientCertificate ? (_, __, ___) => true : _clientCertificateValidation;
```

Comment:
Sneaky, but having a getter return a different value based on some other property smells 👃 .

Try updating the consuming code to check AllowAnyClientCertificate directly.
https://github.com/aspnet/AspNetCore/blob/689b0525095793e6dd35ee3704bc23f52ed43bd0/src/Servers/Kestrel/Core/src/Internal/HttpsConnectionMiddleware.cs#L134

Reply:
Agreed. I think it's better to keep options objects as pocos and simply set "rules" around consuming them (i.e. that if `AllowAnyClientCertificate` is set, `ClientCertificateValidation` will be ignored).

Reply:
Okay, that seems less sneaky